### PR TITLE
T2D-258 file object converter

### PR DESCRIPTION
**Pull request by selvaebi**

### CODE REVIEW

File: metadata-api/src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/entities/File.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+@Entity
+@Table(uniqueConstraints = @UniqueConstraint(columnNames = {"accession", "version"}))
+public class File extends Auditable<Long> {
+
+    public enum Type {
+
+        tsv, tab, bam, bai, cram, crai, vcf, vcf_aggregate, bcf, tabix, wig, bed, gff, fasta, fastq, flatfile,
+        chromosome_list, sample_list, readme_file, phenotype_file, BioNano_native, Kallisto_native, agp,
+        unlocalised_list, info, manifest, other
```
Comment (Zhicheng-Liu) [inline metadata-api/src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/entities/File.java]: There are additional file types for "Run" files:
https://github.com/enasequence/schema/blob/master/src/main/resources/uk/ac/ebi/ena/sra/schema/SRA.run.xsd#L94-L320

Not sure would they ever be used, but just wanted to point it out.

Comment (selvaebi) [reply metadata-api/src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/entities/File.java]: I am not which of the files we gonna use. the analysis file types may contain any old types as well .

File: metadata-load/src/test/java/uk/ac/ebi/ampt2d/metadata/converter/SRAToAMPT2DConverterTest.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SRAToAMPT2DConverterTest {
+
+    @Mock
+    private AnalysisFileType analysisFileType;
+
+    @Test
+    public void testFileConverter() {
```
[the remark this answers is not in the document]
Comment (selvaebi) [reply metadata-load/src/test/java/uk/ac/ebi/ampt2d/metadata/converter/SRAToAMPT2DConverterTest.java]: i think the converter just converts pre validated object , i couldn't think of any failing test cases.

File: metadata-api/src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/entities/File.java (hunk 1 above)
Comment (selvaebi) [reply metadata-api/src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/entities/File.java]: may be once the loading is ready , i could check the filetypes present

Comment (selvaebi) [reply metadata-api/src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/entities/File.java]: > There are additional file types for "Run" files:
> https://github.com/enasequence/schema/blob/master/src/main/resources/uk/ac/ebi/ena/sra/schema/SRA.run.xsd#L94-L320
> 
> Not sure would they ever be used, but just wanted to point it out.

I took the filetypes from Analysis filetypes supported Enum class , so i think others wont be supported or would be tagged as OTHER

**Pull request merged**